src/NodeIPC/Test.hs

```diff
@@ -0,0 +1,37 @@
+module NodeIPC.Test where
+
+import           Cardano.Prelude
+
+import           System.Environment (setEnv, unsetEnv)
+import           System.IO (hClose)
+
+import           NodeIPC.Lib
+import           NodeIPC.Message
+
+--------------------------------------------------------------------------------
+-- Testing
+--------------------------------------------------------------------------------
+
+testIPC :: IO MsgOut
+testIPC = do
+    setEnv "NODE_CHANNEL_FD" "1" -- ???
```

Comment:
You can simply abstract over it. You can just say "you, the caller, choose how you want to pass the environment information to the function, and I will support it". That provides you with the ability to test this and not to think about the way the rest of the system is going to work.
Having said that, think about how you would test the environment functions?
Take a look at the `IOLayer` from our previous project, for example :wink: